internal/controller/apiextensions/claim/configurator.go

```diff
@@ -125,35 +126,37 @@ func ConfigureComposite(_ context.Context, cm resource.CompositeClaim, cp resour
 	}
 
 	claimSpecFilter := xcrd.GetPropFields(wellKnownClaimFields)
-	ucp.Object["spec"] = filter(spec, claimSpecFilter...)
+	cpPatch.(*composite.Unstructured).Object["spec"] = filter(spec, claimSpecFilter...)
```

Comment:
this is ugly. Why not use that type for `cpPatch` directly? A type cast is always suspicious, unguarded even more so.

Reply:
I see why you do that. Still ugly, but I don't see a better way.

Reply:
Me neither, but I see this sort of casting often in this part of the codebase. This gives me a signal that we have just one implementation of `resource.Composite` or `resource.CompositeClaim` interface and perhaps it would be better to use it directly where needed. Using casts mean that we have single data model, and hence there is no much benefit of hiding it behind some interfaces.

Reply:
I have removed type casting in [this commit](https://github.com/crossplane/crossplane/pull/4896/commits/e44abc93a87207d1ffb69e299fa8a52925de6ec4). Let me know what do you think. I like it.